internal/libyaml/api_test.go

```diff
@@ -0,0 +1,527 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package libyaml
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+
+	"go.yaml.in/yaml/v4/internal/testutil/assert"
+)
+
+func TestAPI(t *testing.T) {
+	RunTestCases(t, "api_test.yaml", map[string]TestHandler{
+		"api-new":       runAPINewTest,
+		"api-method":    runAPIMethodTest,
+		"api-panic":     runAPIPanicTest,
+		"api-delete":    runAPIDeleteTest,
+		"api-new-event": runAPINewEventTest,
+	})
+}
+
+func runAPINewTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	obj := createObject(tc.Constructor)
+	runFieldChecks(t, obj, tc.Checks)
+}
+
+func runAPIMethodTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	obj := createObject(tc.Constructor)
+
+	// Run setup if any
+	if tc.Setup != nil {
+		if setupList, ok := tc.Setup.([]interface{}); ok && len(setupList) > 0 {
+			callMethodFromList(obj, setupList, tc.Bytes)
+		}
+	}
+
+	// Call the main method
+	callMethodFromList(obj, tc.Method, tc.Bytes)
+
+	// Run checks
+	runFieldChecks(t, obj, tc.Checks)
+}
+
+func runAPIPanicTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	obj := createObject(tc.Constructor)
+
+	// Run setup if any
+	if tc.Setup != nil {
+		if setupList, ok := tc.Setup.([]interface{}); ok && len(setupList) > 0 {
+			callMethodFromList(obj, setupList, tc.Bytes)
+		}
+	}
+
+	// The main method call should panic
+	// Want can be either a string or a single-element sequence
+	var wantMsg string
+	switch v := tc.Want.(type) {
+	case string:
+		wantMsg = v
+	case []interface{}:
+		msg, ok := v[0].(string)
+		assert.Truef(t, ok, "Want[0] should be string, got %T", v[0])
+		wantMsg = msg
+	default:
+		t.Fatalf("want must be a string or sequence, got %T", tc.Want)
+	}
+	assert.PanicMatchesf(t, wantMsg, func() {
+		callMethodFromList(obj, tc.Method, tc.Bytes)
+	}, "Expected panic: %s", wantMsg)
+}
+
+func runAPIDeleteTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	obj := createObject(tc.Constructor)
+
+	// Run setup if any
+	if tc.Setup != nil {
+		if setupList, ok := tc.Setup.([]interface{}); ok && len(setupList) > 0 {
+			callMethodFromList(obj, setupList, tc.Bytes)
+		}
+	}
+
+	// Call Delete method
+	callMethodFromList(obj, []interface{}{"Delete"}, false)
+
+	// Run checks after delete
+	runFieldChecks(t, obj, tc.Checks)
+}
+
+func runAPINewEventTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	event := createEventFromList(tc.Method, tc.Bytes)
+	runFieldChecks(t, &event, tc.Checks)
+}
+
+// createObject creates a Parser or Emitter based on constructor name
+func createObject(constructor string) interface{} {
```

Comment:
Here and everywhere where you use panic, I would pass t *testing.T as argument and call t.Fatal or t.Fatalf